specification/guestconfiguration/resource-manager/Microsoft.GuestConfiguration/preview/2018-01-20-preview/guestconfiguration.json

```diff
@@ -555,6 +649,58 @@
         }
       },
       "description": "Report for the guest configuration. Report contains information such as compliance status, reason and more."
+    },
+    "SoftwareList": {
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Software"
+          },
+          "description": "The list of software installed on a given virtual machine."
+        }
+      },
+      "description": "The list of software installed on a given virtual machine."
+    },
+    "Software": {
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": false,
+          "$ref": "#/definitions/SoftwareProperties",
+          "description": "Properties of the software."
+        }
+      },
+      "allOf": [
+        {
+        "$ref": "../../common/v1/definitions.json#/definitions/ProxyResource"
+      }
```

Comment:
The base structure taken by this nee resource will be a Proxy resource, it will just get applied to an extension, just like guestConfigurationAssignments.